### PR TITLE
feat: planned route map

### PR DESCRIPTION
**Pull request by itai-rozen**
# Description
after choosing a route & time of route, line of route is displayed on map
2 polylines are loaded: 
- actual route positions based on vehicle locations
- planned route stops based on GTFS API

also, added a map index showing which marker belongs to which route

## screenshots
‏‏לכידה

### CODE REVIEW

**Comment by NoamGaash** [comment]
Hi, welcome to our repo!
Thanks for the contribution. Just let me make sure I understand - it's not about showing the actual / planned ride, it's just about adding the marker. Am I missing something?
By the way, please run `npm run lint:fix` to fix all linting issues

File: src/pages/components/map-related/MapWithLocationsAndPath.tsx [hunk 1]
```diff
@@ -41,17 +48,17 @@ export function MapWithLocationsAndPath({
         onClick={toggleExpanded}
         icon={<ExpandAltOutlined />}
       />
-      <MapContainer center={position.loc} zoom={8} scrollWheelZoom={true}>
+      <MapContainer center={position.loc} zoom={13} scrollWheelZoom={true}>
```
Comment (NoamGaash) [inline src/pages/components/map-related/MapWithLocationsAndPath.tsx:51]: can be reverted

File: src/pages/components/map-related/MapWithLocationsAndPath.tsx [hunk 2]
```diff
@@ -61,15 +68,21 @@ export function MapWithLocationsAndPath({
...
           <Polyline
             key={path.vehicleRef}
             pathOptions={{
               color: getColorByHashString(path.vehicleRef.toString()),
             }}
             positions={path.locations.map(({ lat, lon }) => [lat, lon])}
           />
-        ))}
+        ))
+        } */}
```
Comment (NoamGaash) [inline src/pages/components/map-related/MapWithLocationsAndPath.tsx:85]: code in comment can be removed

**Comment by itai-rozen** [comment]
hey!
so i just replaced the markers with a line that displays the route based on
the 'positions' given. Thats what i thought the issues was addressing.
Correct me if im wrong

‫בתאריך יום א׳, 28 בינו׳ 2024 ב-14:17 מאת ‪Noam Gaash‬‏ <‪
***@***.***‬‏>:‬

**Comment by NoamGaash** [comment]
Actually, this map doesn't show a planned ride - it's showing an actual ride that actually happened. The issue was about adding additional line/map showing the planning

**Comment by itai-rozen** [comment]
aiight i'll work on it:)
dont delete my branch yet

‫בתאריך יום ב׳, 29 בינו׳ 2024 ב-16:09 מאת ‪Noam Gaash‬‏ <‪
***@***.***‬‏>:‬

**Comment by NoamGaash** [comment]
hi, could you please clarify which issue you're solving, and attach an updated screenshot? thanks

**Comment by itai-rozen** [comment]
> # Description
> after choosing a route & time of route, line of route is displayed on map 2 polylines are loaded:
> 
> * actual route positions based on vehicle locations
> * planned route stops based on GTFS API
> 
> also, added a map index showing which marker belongs to which route
> 
> ## screenshots
> ‏‏לכידה

**Pull request merged**